vcs/repository.go

```diff
@@ -140,10 +141,28 @@ type Diff struct {
 	Raw string // the raw diff output
 }
 
+// BranchesOptions specifies options for the list of branches returned by
+// (Repository).Branches.
+type BranchesOptions struct {
+	// BehindAheadBranch specifies a branch name. If set to something other than blank string,
+	// then each returned branch will include a behind/ahead commit counts information
+	// against the specified base branch. If left blank, then branches
+	// will not include that information and their Counts will be nil.
+	BehindAheadBranch string `json:",omitempty" url:",omitempty"`
+}
+
 // A Branch is a VCS branch.
 type Branch struct {
 	Name string
 	Head CommitID
+
+	Counts *BehindAhead `json:",omitempty"` // Counts optionally contains the commit counts relative to specified branch.
+}
+
+// BehindAhead is a set of behind/ahead counts.
+type BehindAhead struct {
+	Behind int
+	Ahead  int
```

Comment:
Maybe these should be made into `uint`, to avoid confusion. Their values are always non-negative. For example:

```
Behind: 4
Ahead: 17
```

That means target branch is 4 commits behind base branch, and 17 commits ahead.